src/services/exchange/exchange.service.ts

```diff
@@ -0,0 +1,49 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { HttpErrorHandler } from '../errors/http-error-handler';
+import { ExchangeResult } from './entities/exchange.entity';
+
+@Injectable()
+export class ExchangeService {
+  // TODO can we depend on the base url instead?
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly httpService: HttpService,
+    private readonly httpErrorHandler: HttpErrorHandler,
+  ) {}
+
+  async convertRates(to: string, from: string): Promise<number> {
+    const exchangeResult = await this.getExchangeResult();
+
+    if (exchangeResult.rates === undefined)
+      throw new InternalServerErrorException(`Exchange rates unavailable`);
+
+    const fromExchangeRate = exchangeResult.rates[from];
+    if (fromExchangeRate === undefined)
+      throw new InternalServerErrorException(
+        `Exchange rate for ${from} is not available`,
+      );
+    const toExchangeRate = exchangeResult.rates[to];
+    if (toExchangeRate === undefined)
+      throw new InternalServerErrorException(
+        `Exchange rate for ${to} is not available`,
+      );
```

Comment:
This can probably be a 4XX – a `404` might be adequate since the fiat code is in the path. We can optionally return the rates available for conversion to the clients.

However this assumes that we do not default to `0` in case we have an error (separate ongoing discussion).